### PR TITLE
:fire: add dockerised pa11y-dashboard

### PR DESCRIPTION
**Pull request by ajthomascouk**

### CODE REVIEW

**Comment by ajthomascouk** [comment]
Need to update the [README.md](https://github.com/nhsuk/pa11y-dashboard/blob/master/README.md)

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+0.1.0 / 2017-09-20
```
Comment (st3v3nhunt) [inline CHANGELOG.md:1]: Hang off on adding the date until it has been released. Use `TBC` or `TBA` as a place holder.

File: rancher-config/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+version: '2'
+
+services:
+    image: "nhsuk/pa11y-dashboard:${DOCKER_IMAGE_TAG}"
+    database:
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: There is no need to have the database specified here.

**Comment by ajthomascouk** [comment]
Commits will need squashing..

**Comment by ajthomascouk** [comment]
Last 5 commits need to be squashed.

File: rancher-config/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        environment:
+            WEBSERVICE_DATABASE: mongodb:${MONGODB_URL}
+        links:
+            - database
+    labels:
+      traefik.enable: true
+      traefik.frontend.rule: $TRAEFIK_RULE
+      traefik.backend: ${RANCHER_STACK_NAME}-pa11y-dashboard
+      traefik.port: 3000
+      prometheus.port: 3000
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: There isn't going to be any metrics for prometheus to scrape in this image.

File: rancher-config/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            - 3000:3000
+        environment:
+            WEBSERVICE_DATABASE: mongodb:${MONGODB_URL}
+        links:
+            - database
+    labels:
+      traefik.enable: true
+      traefik.frontend.rule: $TRAEFIK_RULE
+      traefik.backend: ${RANCHER_STACK_NAME}-pa11y-dashboard
+      traefik.port: 3000
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: I think this will need to be `3333`

File: rancher-config/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+services:
+    image: nhsuk/pa11y-dashboard:${DOCKER_IMAGE_TAG}
+    database:
+        image: mongo:3.3
+    dashboard:
+        build:
+            context: .
+        ports:
+            - 3333:3333
+            - 3000:3000
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: I don't think you need to expose port `3000` here.

File: rancher-config/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        environment:
+            WEBSERVICE_DATABASE: mongodb:${MONGODB_URL}
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: Is `MONGODB_URL` in vault? It'll need to be for a value to be set and used.

File: rancher-config/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+version: '2'
+
+services:
+    image: nhsuk/pa11y-dashboard:${DOCKER_IMAGE_TAG}
+    database:
```
Comment (st3v3nhunt) [inline rancher-config/docker-compose.yml]: I don't think this `database` section is required as it isn't going to be running a local db when running in rancher.

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    database:
+        image: mongo:3.3
+
+    dashboard:
+        build:
+            context: .
+
+        ports:
+            - 3333:3333
+            - 3000:3000
```
Comment (st3v3nhunt) [inline docker-compose.yml:13]: What does port 3000 expose?

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,19 @@
+FROM node:4
```
Comment (st3v3nhunt) [inline Dockerfile:1]: It would be good to see if this runs on the latest version of node (`8.6.0`) when it is all running.

**Review by st3v3nhunt — CHANGES_REQUESTED** [review]
Some questions and suggestions.

File: rancher-config/docker-compose.yml (hunk 6 above)
Comment (ajthomascouk) [reply rancher-config/docker-compose.yml]: Yeah

File: docker-compose.yml (hunk 8 above)
Comment (ajthomascouk) [reply docker-compose.yml:13]: MongoDB

File: CHANGELOG.md (hunk 1 above)
Comment (ajthomascouk) [reply CHANGELOG.md:1]: Yeah good point

File: rancher-config/docker-compose.yml (hunk 2 above)
Comment (ajthomascouk) [reply rancher-config/docker-compose.yml]: Removed.

File: rancher-config/docker-compose.yml (hunk 5 above)
Comment (ajthomascouk) [reply rancher-config/docker-compose.yml]: Removed.

File: rancher-config/docker-compose.yml (hunk 4 above)
Comment (ajthomascouk) [reply rancher-config/docker-compose.yml]: Changed

File: rancher-config/docker-compose.yml (hunk 3 above)
Comment (ajthomascouk) [reply rancher-config/docker-compose.yml]: Removed `prometeus`

**Comment by st3v3nhunt** [comment]
Superseded by #4

**Pull request closed without merging**